### PR TITLE
Chore: Add `no-duplicate-case` eslint rule and fix violations

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**
Adds `no-duplicate-case` rule and fixes problems

**Why do we need this feature?**
In Vite investigation (https://github.com/grafana/grafana/pull/80664), the build warns about duplicated switch cases. This is something we should fix, and can sort separately to the Vite migration anyway

**Who is this feature for?**
UI devs

### CODE REVIEW

File: public/app/core/utils/navBarItem-translations.ts [hunk 1]
```diff
@@ -153,8 +153,6 @@ export function getNavTitle(navId: string | undefined) {
       return t('nav.slo.title', 'SLO');
     case 'plugin-page-k6-app':
       return t('nav.k6.title', 'Performance');
-    case 'monitoring':
-      return t('nav.observability.title', 'Observability');
```
Comment (tomratcliffe) [inline public/app/core/utils/navBarItem-translations.ts:157]: This is duplicated by another case higher up, which has the same display value (`Observability`)

File: public/app/features/expressions/ExpressionQueryEditor.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ function useExpressionsCache() {
       // We want to use the same value for Reduce, Resample and Threshold
       case ExpressionQueryType.reduce:
       case ExpressionQueryType.resample:
-      case ExpressionQueryType.resample:
+      case ExpressionQueryType.threshold:
```
Comment (jackw) [inline public/app/features/expressions/ExpressionQueryEditor.tsx:47]: Did this catch a bug? 🤔

**Review by jackw — APPROVED** [review]
LGTM! 🚀

File: public/app/features/expressions/ExpressionQueryEditor.tsx (hunk 2 above)
Comment (tomratcliffe) [reply public/app/features/expressions/ExpressionQueryEditor.tsx:47]: I think so - but I wasn't able to get into a breakpoint in this code so I can't say for sure 🤷 Regardless, I think it should go in

**Pull request merged**